src/legacy/core_plugins/kibana/public/dashboard/dashboard_app_controller.tsx

```diff
@@ -319,6 +319,14 @@ export class DashboardAppController {
             kbnUrl.removeParam(DashboardConstants.ADD_VISUALIZATION_TO_DASHBOARD_MODE_PARAM);
             kbnUrl.removeParam(DashboardConstants.NEW_VISUALIZATION_ID_PARAM);
           }
+          if ($routeParams[DashboardConstants.NEW_LENS_VISUALIZATION_ID_PARAM]) {
```

Comment:
I wonder if it would make sense to generalize this URL directly. So that you can pass the embeddable type and the embeddable id via a URL instead of creating a specific case for each embeddable here.

Also -- especially since we don't have a good "URL service" to navigate between apps right now -- I like an idea that the observability plugin is taking, that they create short URLs for those kind of links, and then just internally forward to the long URL, so that those short URLs can stay forever the same even if something internally needs to change. So I wonder if we should have a URL in the format `/dashboard/add/<dashboard-id>/<embeddable-type>/id/<embeddable-id>`, and make this forward to the appropriate methods internally. (I btw added the `id` here explicitly, since I think we later want to be able to add also embeddables by their full embeddable config instead of saved embeddables via that way.)

@majagrubic what are your thoughts on that?

Reply:
I like this idea. But to be sure I understand what you are saying - writing the URL in such a way would allow for a more generic way of adding embeddables, and solves the problem of having Lens reference inside Dashboard. The problem of "not having a good URL service to navigate between apps" still remains, and the general idea of updating `window.location` still remains.

Reply:
Additionally, we'd need to handle the case when there is no `dashboard-id`, as the dashboard hasn't yet been saved.

Reply:
Yes I totally agree. This does not solve the generic problem of us needing to build proper URL services in the long run. So this is just an easier way to keep URLs stable before we have those kind of URL services. (And even they might want to use those short URLs in the long run.)

Reply:
I made a code change, so that `embeddableType` and `embeddableId` are more generic, but still passed as query params. Let me know if this is more along the lines what you were thinking.